src/test/java/org/openrewrite/java/migrate/guava/PreferJavaUtilOptionalTest.java

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright 2021 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.guava;
+
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junitpioneer.jupiter.ExpectedToFail;
+import org.openrewrite.Issue;
+import org.openrewrite.config.Environment;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+@Issue("https://github.com/openrewrite/rewrite-migrate-java/issues/197")
+class PreferJavaUtilOptionalTest implements RewriteTest {
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(
+            Environment.builder()
+              .scanRuntimeClasspath("org.openrewrite.java.migrate.guava")
+              .build()
+              .activateRecipes("org.openrewrite.java.migrate.guava.PreferJavaUtilOptional")
+          )
+          .parser(JavaParser.fromJavaVersion().classpath("guava"));
+    }
+
+    @Test
+    void absentToEmpty() {
+        //language=java
+        rewriteRun(java("""
+          import com.google.common.base.Optional;
+
+          class A {
+              Optional<String> foo() {
+                  return Optional.absent();
+              }
+          }
+          """, """
+          import java.util.Optional;
+
+          class A {
+              Optional<String> foo() {
+                  return Optional.empty();
+              }
+          }
+          """));
+    }
+
+    @Test
+    void orNullToOrElseNull() {
+        // Comparison to java.util.Optional: this method is equivalent to Java 8's Optional.orElse(null).
+        //language=java
+        rewriteRun(java("""
+              import com.google.common.base.Optional;
+
+              class A {
+                  String foo(Optional<String> optional) {
+                      return optional.orNull();
+                  }
+              }
+              """, """
+              import java.util.Optional;
+
+              class A {
+                  String foo(Optional<String> optional) {
+                      return optional.orElse(null);
```

Comment:
It looks like the type attribution is slightly messed up by the recipe. In the following screenshot you see the method type of the method invocation of the `orElse()` method. I was slightly surprised by the `String` return type, but I think this is normal for OpenRewrite as it uses `JavaType.Method` both for method declarations and method invocations and they are slightly different in this respect. What is clearly missing, however, is the parameter type.
<img width="824" alt="image" src="https://user-images.githubusercontent.com/408765/227860850-1dd946ed-48e0-451c-aa8e-def0de387021.png">